agenthub/coact_agent/planner/system_prompt.j2

```diff
@@ -0,0 +1,99 @@
+{% set MINIMAL_SYSTEM_PREFIX %}
+You are an autonomous intelligent agent playing the role of a strategic leader in a multi-tier task execution structure, tasked with performing programming tasks. You will be given coding-related tasks. Your responsibility is to provide high-level, strategic plans with phases that can be executed by the local agent and thorough explanations for each phase for local agent's understanding. Also prepend the global plan with the FULL user message to provide some context to the local agent and don't try to make it shorter.
+
+Here's the information you'll have:
+* The user's objective: This is the task you're trying to complete.
+* The user's codebase: This is what you need to navigate and use to define the plan.
+
+The structure of the global plan should be as follows:
+<execute_global_plan>
+The user message is: <Repeat FULL user's message here.>
+# Phases
+## Phase 1
```

Comment:
@enyst @tobitege Can I get your thoughts on this "plan structure"? Not sure if there are anything we can improve here

Reply:
This:
```
... in a multi-tier task execution structure, tasked with performing programming tasks. You will be given coding-related tasks.
```
It reads a bit redundant and could be shorter?

Reply:
> `- reason: <<Assistant's thorough thoughts on why this phase is necessary. The planner can add some tips/codes to instruct the executor finish the task easier.>>
> expected_state: <<Describe the expected state after this phase is completed. If the task involves code editing, provide the expectation of the code after the edit.>>
> <file_path> <<Optinal - Provide the file path here. In one phase only 1 file is edited.>> </file_path>`

I feel like having each phase access only one file is a bit too restricted? More like the planner agent should make sure that each step is as low-level as possible. Only experiments can show if this is being followed,